src/core/UBPreferencesController.h

```diff
@@ -152,20 +155,22 @@ private slots:
 
 private:
     QList<QPushButton*> mScreenLabels;
-    QValidator* mValidator;
+    UBStringListValidator* mValidator;
 };
 
 class UBStringListValidator : public QValidator
```

Comment:
For clarity/accuracy reasons, this class should be more explicit and be renamed `UBScreenNamesValidator`

Reply:
For this and everything below see also the discussion with @habruening above. I made some effort to make the `UBStringListValidator` completely unaware of what it validates. It just checks whether the input string can be split into trimmed tokens and that then each of the tokens is contained in the validation string list and occurs at most once. 

E.g. I had not to change anything in the validator when moving from string names to string indexes. It is completely unaware of that. And as with the string splitter above, it could also be used on another completely unrelated input field containing e.g. a comma separated list of options for something completely different. So again I would argue that it should not have "screen" as part of its name or any of its members.